plot_continuous_monitor.py

```diff
@@ -129,106 +94,66 @@ def plot_continuous_monitor(filename, open_graphs=False):
-  for line in open(os.path.join(scripts_dir, "plot_disk_utilization_base.gp"), "r"):
-    new_line = line.replace("__OUT_FILENAME__", "%s_disk_utilization.pdf" % filename).replace(
-      "__NAME__", out_filename)
-    disk_plot_file.write(new_line)
-  disk_plot_file.close()
 
-  xvdf_plot_output_name = plot_single_disk(filename, out_filename, "xvdf", ["xvdb"], scripts_dir)
-  xvdb_plot_output_name = plot_single_disk(filename, out_filename, "xvdb", ["xvdf"], scripts_dir)
-
-  monotasks_plot_filename = "%s_monotasks.gp" % filename
-  monotasks_plot_file = open(monotasks_plot_filename, "w")
-  for line in open(os.path.join(scripts_dir, "plot_monotasks_base.gp"), "r"):
-    new_line = line.replace("__OUT_FILENAME__", "%s_monotasks.pdf" % filename).replace(
-      "__NAME__", out_filename)
-    monotasks_plot_file.write(new_line)
-  monotasks_plot_file.close()
-
-  memory_plot_filename = "%s_memory.gp" % filename
-  memory_plot_file = open(memory_plot_filename, "w")
-  for line in open(os.path.join(scripts_dir, "plot_memory_base.gp"), "r"):
-    new_line = line.replace("__OUT_FILENAME__", "%s_memory.pdf" % filename).replace(
-      "__NAME__", out_filename)
-    memory_plot_file.write(new_line)
-  memory_plot_file.close()
-
-  subprocess.check_call("gnuplot %s" % utilization_plot_filename, shell=True)
-  subprocess.check_call("gnuplot %s" % monotasks_plot_filename, shell=True)
-  subprocess.check_call("gnuplot %s" % disk_plot_filename, shell=True)
-  subprocess.check_call("gnuplot %s" % memory_plot_filename, shell=True)
+def get_util_for_disk(disk_utils, disk):
+  """
+  Returns the disk utilization metrics for the specified disk, given the
+  utilization information for all disks, or None if the desired disk cannot be
+  found.
+  """
+  for disk_util in disk_utils:
+    if disk in disk_util:
+      return disk_util[disk]
+  return None
 
-  if (open_graphs):
-    subprocess.check_call("open %s_utilization.pdf" % filename, shell=True)
-    subprocess.check_call("open %s_monotasks.pdf" % filename, shell=True)
-    subprocess.check_call("open %s_disk_utilization.pdf" % filename, shell=True)
-    subprocess.check_call("open %s_memory.pdf" % filename, shell=True)
-    subprocess.check_call("open %s" % xvdf_plot_output_name, shell=True)
-    subprocess.check_call("open %s" % xvdb_plot_output_name, shell=True)
 
 def parse_args():
   parser = argparse.ArgumentParser(description="Plots Spark continuous monitor logs.")
   parser.add_argument(
     "-f", "--filename", help="The path to a continuous monitor log file.", required=True)
   parser.add_argument(
-    "-o",
-    "--open-graphs",
-    action="store_true",
-    default=False,
-    dest="open_graphs",
-    help="Whether to open the resulting graph PDFs.")
+    "-o", "--open_graphs", help="open generated graphs", action="store_true")
+  parser.add_argument(
+    "-g", "--gnuplot", help="generate graphs with gnuplot", action="store_true")
+  parser.set_defaults(gnuplot=False, open_graphs=False)
```

Comment:
Can you leave the old default mechanism (makes the args easier to read, since the default is close to the definition)